src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/SearchOptionsFactoryTests.cs

```diff
@@ -110,6 +110,22 @@ public void GivenACount_WhenCreated_ThenCorrectMaxItemCountShouldBeSet()
             Assert.Equal(5, options.MaxItemCount);
         }
 
+        [Fact]
+        public void GivenACountWithValueZero_WhenCreated_ThenCorrectMaxItemCountShouldBeSet()
+        {
+            const ResourceType resourceType = ResourceType.Encounter;
+            var queryParameters = new[]
+            {
+               Tuple.Create("_count", "0"),
+            };
+
+            SearchOptions options = CreateSearchOptions(
+            resourceType: resourceType.ToString(),
+            queryParameters: queryParameters);
+
+            Assert.NotNull(options);
```

Comment:
Can you add a check for the summary being count?

Reply:
Done

Reply:
I was meaning to check the options.Summary field to ensure it got populated.

Reply:
As discussed added check for options.CountOnly field